File: proddata/data/3sum/Python/Solution.py
class Solution:
    def threeSum(self, nums):
        if len(nums) < 3:
            return []
        nums.sort()
        ans = []
        for cur_ind in range(len(nums)-1):
            if cur_ind > 0 and nums[cur_ind] == nums[cur_ind - 1]:
                continue
            left, right = cur_ind + 1, len(nums)-1
            while left < right:
                threesum = nums[cur_ind] + nums[left] + nums[right]
                if threesum > 0:
                    right -= 1
                elif threesum < 0:
                    left += 1
                else:
                    ans.append([nums[cur_ind], nums[left], nums[right]])
                    left += 1
                    while nums[left] == nums[left - 1] and left < right:
                        left += 1
        return ans

File: proddata/data/3sum/PHP/Solution.php
<?php
namespace Sol;
class Solution1 {
    public $resHash;
    /**
     * @param Integer[] $nums
     * @return Integer[][]
     */
    function threeSum($nums) {
        sort($nums);
        $res = [];
        
        for($i = 0; $i < count($nums); $i++) {
            $l = $i+1;
            $r = count($nums)-1;
            while($l < $r) {
                $sum = $nums[$i]+$nums[$l]+$nums[$r];
                if($sum === 0) {
                    if($resHash[$nums[$i].$nums[$l].$nums[$r].""] !== 1) {
                        array_push($res, [$nums[$i],$nums[$l],$nums[$r]]);
                        $resHash[$nums[$i].$nums[$l].$nums[$r].""] = 1;
                    }
                    $l += 1;
                    $r -= 1;
                } else if($sum < 0) {
                    $l += 1;
                } else {
                    $r -= 1;
                }
            }
        }
        return $res;
    }
}
?>